proto/storage/base_image_repository.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package storage;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "./storage;storage";
+option java_package = "io.stackrox.proto.storage";
+
+// BaseImageRepository configures automatic tag discovery for a base image repository.
+// Defines which repository to watch, what tag patterns to match, and health tracking.
+// Next tag: 8
+message BaseImageRepository {
+  // Unique identifier for this repository pattern configuration.
+  string id = 1; // @gotags: sql:"pk,type(uuid)"
+
+  // Full repository path including registry (e.g., "registry.redhat.io/ubi8/ubi").
+  // Must be unique - one tag pattern per repository.
```

Comment:
If this is unique, The API would need to support "Update" operation(PUT). Is it the plan? I remembered in some meeting, people say it should be "create" -> "delete"(POST) model. Can you clarify? I have no objection to this just to clarify because the API is defined as POST for now.

Reply:
> If this is unique, The API would need to support "Update" operation(PUT). 

What is the use case for PUT? On my view:

1. POST: Create-only. Add a new repository path, fails if repository a path already exists.
2. DELETE: Delete-only. Deletes if an entry with the same repository path exists, fail if not found.

That supports the "create -> delete" model.